libcodechecker/util.py

```diff
@@ -172,6 +172,49 @@ def get_default_workspace():
     return workspace
 
 
+def __strip_protocol_from_url(url):
+    """
+    Strip the PROTOCOL specifier from an URL string and return it, along with
+    the remainder of the URL.
+    """
+    if '://' not in url:
+        return None, url
+
+    parts = url.split('://', 1)
+    protocol = parts[0]
+    url = url.replace(parts[0] + '://', '').lstrip('/').rstrip('/')
+
+    return protocol, url
+
+
+def expand_whole_protocol_and_port(protocol=None, port=None):
+    """
+    Calculate a full protocol and port value from the possibly None protocol
+    and port values given. This method helps to default port numbers to
+    connection protocols understood by CodeChecker.
+    """
+
+    proto, portnum = None, None
+    if protocol:
```

Comment:
Does it make sense to call this method without protocol? Maybe `'http'` would be a better defaul than none and would simplify the code?

Also I am confused by this convoluted logic. If protocol is there the port for http is 80 otherwise 8001? Why is is the distinction?

Reply:
CodeChecker's default port is `8001`.

Reply:
So if I call this with `None`, I will get `http, 8001`, if I call this with `http`, I will get `http, 80`.  I find this behavior confusing.

Reply:
This behaviour is not confusing if you think about where this function is used.
By default, CodeChecker starts on port 8001. So users who don't use any specialisation and just use defaults, it will work.

If you have a server outside local host, most users will usually end up copy-pasting browser URLs, or URLs from emails, or typing them in by hand.
If you copy an URL that uses HTTP on the standard port, the `:80` string is not copied by any browser of yesterday or today.

We can't make CodeChecker start on port 80 as it would require root privileges.

Reply:
Also, most users don't type in `http://` as it is the default while browsing. Which is why a `None, None` invocation must be supported to set up these defaults.

Reply:
I think http://  defaulting to 80, none defaulting to 8001, and https://defaulting to 443 is fine behaviour. 8001 is okay as this is the default prot the server binds to.